crates/extensions/c8y_mapper_ext/src/operations/upload.rs

```diff
@@ -0,0 +1,105 @@
+use c8y_http_proxy::messages::CreateEvent;
+use camino::Utf8Path;
+use mime::Mime;
+use std::collections::HashMap;
+use tedge_api::entity_store::EntityExternalId;
+use tedge_api::mqtt_topics::OperationType;
+use tedge_api::workflow::GenericCommandState;
+use tedge_config::AutoLogUpload;
+use tedge_uploader_ext::ContentType;
+use tedge_uploader_ext::FormData;
+use tedge_uploader_ext::UploadRequest;
+use tedge_uploader_ext::UploadResult;
+use time::OffsetDateTime;
+use url::Url;
+
+use crate::error::ConversionError;
+
+use super::OperationHandler;
+
+impl OperationHandler {
+    #[allow(clippy::too_many_arguments)]
+    pub(crate) async fn upload_file(
+        &self,
+        external_id: &EntityExternalId,
+        file_path: &Utf8Path,
+        file_name: Option<String>,
+        mime_type: Option<Mime>,
+        cmd_id: &str,
+        event_type: String,
+        event_text: Option<String>,
+    ) -> Result<(Url, UploadResult), ConversionError> {
+        let create_event = CreateEvent {
+            event_type: event_type.clone(),
+            time: OffsetDateTime::now_utc(),
+            text: event_text.unwrap_or(event_type),
+            extras: HashMap::new(),
+            device_id: external_id.into(),
+        };
+
+        let mut c8y_http_proxy = self.http_proxy.lock().await.clone();
```

Comment:
As `C8YHttpProxy` the use of `Arc<Mutex<C8YHttpProxy>>` can be simply removed.

```suggestion
        let mut c8y_http_proxy = self.http_proxy.clone();
```

Reply:
fixed in f72e3c4f